lib/_stream_writable.js

```diff
@@ -590,6 +591,10 @@ Writable.prototype.end = function(chunk, encoding, cb) {
   // Ignore unnecessary end() calls.
   if (!state.ending)
     endWritable(this, state, cb);
+  else
+    process.nextTick(() => {
+      cb(new ERR_STREAM_DESTROYED('end'));
+    });
```

Comment:
I would suggest:

```js
  process.nextTick(cb, new ERR_STREAM_DESTROYED('end'));
```

because a) that way the stack trace for the error is more useful by default (`nextTick` remove context) & b) it’s more efficient to pass arguments to `process.nextTick()` rather than using closures (or at least it should be) :slightly_smiling_face:

Reply:
I like!